api/src/main/java/org/commonjava/indy/conf/IndyLoggingConfig.java

```diff
@@ -0,0 +1,164 @@
+    {
+        return new File( IndyConfigInfo.CONF_INCLUDES_DIR, "logging.conf" ).getPath();
+    }
+
+    @Override
+    public InputStream getDefaultConfig()
+    {
+        return Thread.currentThread().getContextClassLoader().getResourceAsStream( "default-logging.conf" );
+    }
+
+    @Override
+    public synchronized void parameter( final String name, final String value )
+            throws ConfigurationException
+    {
+        switch ( name )
+        {
+            case HOST_NAME_KEY:
+            {
+                this.hostname = value;
+                break;
+            }
+            case BUILD_COMMIT_KEY:
+            {
+                this.openshiftBuildCommit = value;
+                break;
+            }
+            case BUILD_NAME_KEY:
+            {
+                this.openshiftBuildName = value;
+                break;
+            }
+            case BUILD_NAMESPACE_KEY:
+            {
+                this.openshiftBuildNamespace = value;
+                break;
+            }
+            default:
+            {
+                throw new ConfigurationException(
+                        "Invalid value: '{}' for parameter: '{}', they are not accepted for section: '{}'.", value,
+                        name, SECTION_NAME );
+            }
+
+        }
+    }
+
+    public String getHostname()
+    {
+        return hostname == null ? System.getenv( HOST_NAME_KEY ) : hostname;
+    }
+
+    @ConfigName( HOST_NAME_KEY )
+    public void setHostname( String host )
+    {
+        this.hostname = host;
+    }
+
+    public String getOpenshiftBuildCommit()
+    {
+        return openshiftBuildCommit == null ? System.getenv( BUILD_COMMIT_KEY ) : openshiftBuildCommit;
+    }
+
+    @ConfigName( BUILD_COMMIT_KEY )
+    public void setOpenshiftBuildCommit( String buildCommit )
+    {
+        this.openshiftBuildCommit = buildCommit;
+    }
+
+    public String getOpenshiftBuildName()
+    {
+        return openshiftBuildName == null ? System.getenv( BUILD_NAME_KEY ) : openshiftBuildName;
+    }
+
+    @ConfigName( BUILD_NAME_KEY )
+    public void setOpenshiftBuildName( String buildName )
+    {
+        this.openshiftBuildName = buildName;
+    }
+
+    public String getOpenshiftBuildNamespace()
+    {
+        return openshiftBuildNamespace == null ? System.getenv( BUILD_NAMESPACE_KEY ) : openshiftBuildNamespace;
+    }
+
+    @ConfigName( BUILD_NAMESPACE_KEY )
+    public void setOpenshiftBuildNamespace( String buildNamespace )
+    {
+        this.openshiftBuildNamespace = buildNamespace;
+    }
+
+    public Map<String, String> getEnvars()
+    {
+        Map<String, String> result = new HashMap<>();
+        result.put( HOST_NAME_KEY, getHostname() );
+        result.put( BUILD_COMMIT_KEY, getOpenshiftBuildCommit() );
+        result.put( BUILD_NAME_KEY, getOpenshiftBuildName() );
+        result.put( BUILD_NAMESPACE_KEY, getOpenshiftBuildNamespace() );
+        return result;
```

Comment:
I would suggest we don't config those. It will be a burden before deploying a new instance. And, I wonder whether we can get it before building the image? i.e., how to get openshift buildname/commit before we make the build? I guess the config doesn't know the info. I might miss something but this sounds strange to me.

Reply:
I will also confirm under NOS-1658, thanks.